sway/input/input-manager.c

```diff
@@ -785,7 +846,18 @@ void input_manager_configure_xcursor(void) {
 
 struct input_config *input_device_get_config(struct sway_input_device *device) {
 	struct input_config *wildcard_config = NULL;
+	struct input_config *type_config = NULL;
 	struct input_config *input_config = NULL;
+	const char *device_type = input_device_get_type(device);
+
+	for (int i = 0; i < config->input_type_configs->length; ++i) {
```

Comment:
Can you move this below the input config loop? If a device specific config is found, there is no reason to check the type configs. Additionally with this change, the `type_config` variable can be removed and just return the type config when found. The `device_type` variable declaration can be moved to be just before the loop as well (no blank line).

Reply:
Looks like this still needs to be done. Other than that, looks good to me